packages/browser/src/index.ts

```diff
@@ -25,4 +45,33 @@ export default class LogtoClient {
   constructor(logtoConfig: LogtoConfig) {
     this.logtoConfig = logtoConfig;
   }
+
+  public async signIn(redirectUri: string) {
+    const { clientId, endpoint, requester, resources, scopes: originalScopes } = this.logtoConfig;
+    const discoveryEndpoint = new URL(discoveryPath, endpoint);
+    this.oidcConfig = await fetchOidcConfig(discoveryEndpoint.toString(), requester);
```

Comment:
I prefer to fetch OIDC config at first.
And I'm also good with fetch OIDC config when needed.
Finally i decided to conform to the majority,
since Swift & Kotlin SDK fetch OIDC config when needed (`oidcConfig` property is optional in LogtoClient) for now.

@logto-io/eng any idea?

Reply:
Updated.

```ts
  public async signIn(redirectUri: string) {
    const { clientId, resources, scopes: customScopes } = this.logtoConfig;
    const oidcConfig = await this.getOidcConfig();

    const { authorizationEndpoint } = oidcConfig;
    ……
  }

  private async getOidcConfig() {
    if (!this.oidcConfig) {
      const { endpoint, requester } = this.logtoConfig;
      const discoveryEndpoint = new URL(discoveryPath, endpoint);
      this.oidcConfig = await fetchOidcConfig(discoveryEndpoint.toString(), requester);
    }

    return this.oidcConfig;
  }
```

See [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1644809320215889) .